matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt

```diff
@@ -63,96 +70,128 @@ internal class SyncResponseHandler @Inject constructor(
         val isInitialSync = fromToken == null
         Timber.v("Start handling sync, is InitialSync: $isInitialSync")
 
-        measureTimeMillis {
-            if (!cryptoService.isStarted()) {
-                Timber.v("Should start cryptoService")
-                cryptoService.start()
+        val relevantPlugins = metricPlugins.filterIsInstance<SyncDurationMetricPlugin>()
+        measureMetric(relevantPlugins) {
```

Comment:
I think we should try to improve readability of this code, what about extracting all the steps in dedicated small methods. And we could also avoid adding comments using good naming for these methods. For example:
```
measureMetric(relevantPlugins) {
    startCryptoService()
    handleToDevice()
    ...
}
```